packages/core/src/pubsub/__tests__/pubsub-ratelimit.test.ts

```diff
@@ -111,7 +119,7 @@ describe('pubsub with queries rate limited', () => {
     for (let i = 1; i < firstElements.length; i++) {
       const current = firstElements[i]
       const previous = firstElements[i - 1]
-      const difference = current.valueOf() - previous.valueOf()
+      const difference = floorSecond(current) - floorSecond(previous)
```

Comment:
if the "seconds" can actually wind up being 999ms apart instead of a full second, then I guess there still is a rare condition where this can fail, if the first event happens exactly at a second boundary (milleseconds of 000) and the second event happens 999ms later, they will both round down to the same second and have a difference of 0.

Reply:
maybe instead of rounding down we can just have an explicit "grace window" of say... 3ms?

Reply:
so something like:
```
// because sleep intervals aren't perfectly reliable and can be off by a millisecond or two, we
// give a 3 millisecond "buffer" to the time comparison to prevent the test from failing sporadically.
const GRACE_WINDOW = 3
expect(difference >= ONE_SECOND - GRACE_WINDOW)
```